charts/jenkins/templates/jenkins-controller-ingress.yaml

```diff
@@ -46,8 +54,15 @@ spec:
   - http:
       paths:
       - backend:
+{{- if semverCompare ">=1.19-0" $kubeTargetVersion -}}
+          service:
+            name: {{ template "jenkins.fullname" . }}
+            port:
+              number: {{ .Values.controller.servicePort }}
+{{- else -}}
           serviceName: {{ template "jenkins.fullname" . }}
           servicePort: {{ .Values.controller.servicePort }}
+{{- end }}
```

Comment:
I like this part. Could you add a unit test, which verifies the templating?
See https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/tests/jenkins-controller-ingress-test.yaml

https://github.com/quintush/helm-unittest/blob/master/DOCUMENT.md#test-suite explain how one can set capabilities.